typescript/src/actions/evm/submitSwapTransaction.ts

```diff
@@ -0,0 +1,145 @@
+import { concat, numberToHex, size } from "viem";
+
+import { sendTransaction } from "./sendTransaction.js";
+
+import type {
+  CdpOpenApiClientType,
+  CreateSwapResponse,
+  SendEvmTransactionBodyNetwork,
+} from "../../openapi-client/index.js";
+import type { Address, Hex } from "../../types/misc.js";
+import type { TransactionRequestEIP1559 } from "viem";
+
+/**
+ * Options for submitting a swap transaction.
+ */
+export interface SubmitSwapTransactionOptions {
+  /**
+   * The address of the account that will execute the swap.
+   */
+  address: Address;
+
+  /**
+   * The network to execute the swap on (e.g., "ethereum", "ethereum-sepolia", "base").
+   */
+  network: SendEvmTransactionBodyNetwork;
+
+  /**
+   * The swap transaction data returned by the createSwap method.
+   */
+  swap: CreateSwapResponse;
+
+  /**
+   * Optional idempotency key for the request.
+   */
+  idempotencyKey?: string;
+}
+
+/**
+ * Result of submitting a swap transaction.
+ */
+export interface SubmitSwapTransactionResult {
+  /**
+   * The transaction hash of the submitted swap.
+   */
+  transactionHash: Hex;
+}
+
+/**
+ * Submits a swap transaction to the blockchain.
+ * Handles any permit2 signatures required for the swap.
+ *
+ * @param {CdpOpenApiClientType} client - The client to use for submitting the swap.
+ * @param {SubmitSwapTransactionOptions} options - The options for the swap submission.
+ * @param {Address} options.address - The address of the account that will execute the swap.
+ * @param {SendEvmTransactionBodyNetwork} options.network - The network to execute the swap on.
+ * @param {CreateSwapResponse} options.swap - The swap transaction data returned by the createSwap method.
+ * @param {string} [options.idempotencyKey] - Optional idempotency key for the request.
+ *
+ * @returns {Promise<SubmitSwapTransactionResult>} A promise that resolves to the transaction hash.
```

Comment:
I think this should probably be "SendSwapTransactionResult" since the base method is "SendTransaction"